hs_core/templates/pages/genericresource.html

```diff
@@ -772,9 +773,9 @@ <h4 class="modal-title">Manage access</h4>
 
                                                 <ul id="list-roles" class="dropdown-menu"
                                                     aria-labelledby="roles_list">
-                                                    <li><a data-role="view">Can view</a></li>
-                                                    <li><a data-role="edit">Can edit</a></li>
-                                                    <li><a data-role="owner">Is owner</a></li>
+                                                    <li {% if not is_view_user and not is_edit_user and not is_owner_user %}class="disabled"{% endif %}><a data-role="view">Can view</a></li>
+                                                    <li {% if not is_edit_user and not is_owner_user %}class="disabled"{% endif %}><a data-role="edit">Can edit</a></li>
+                                                    <li {% if not is_owner_user %}class="disabled"{% endif %}><a data-role="owner">Is owner</a></li>
```

Comment:
@mauriel133 "Manage access" button only shows up when the requesting user clicks on edit button to get to edit mode, which means the requesting user has at least edit permission to the resource, so I think the check for "Can view" and "Can edit" roles are not needed here. I think the only check needed here is for "Is owner" role to make sure only owners of the resource can add other owners.